src/io/overlapped.cr

```diff
@@ -49,17 +49,56 @@ module IO::Overlapped
     end
   end
 
+  def self.wait_queued_completions(timeout)
+    overlapped_entries = uninitialized LibC::OVERLAPPED_ENTRY[1]
+
+    if timeout > UInt64::MAX
+      timeout = LibC::INFINITE
```

Comment:
Not relevant right now probably as this is the single threaded version of crystal (right?), but at some point this would probably would be clamped downwards towards a short duration rather than towards infinity. This because otherwise if a fiber is woken up from waiting on a channel, then if the fiber is assigned to a thread that is in an infinite wait won't process the woken fiber until the completion return.